interp/src/interpreter/steppers/control_interpreter.rs

```diff
@@ -591,38 +733,123 @@ impl<'a> Interpreter for WhileInterpreter<'a> {
             vec![]
         }
     }
+
+    fn get_mut_env(&mut self) -> MutStateView<'_, 'outer> {
+        if let Some(cond) = &mut self.cond_interp {
+            cond.get_mut_env()
+        } else if let Some(body) = &mut self.body_interp {
+            body.get_mut_env()
+        } else if let Some(term) = &mut self.terminal_env {
+            term.into()
+        } else {
+            unreachable!("Invalid internal state for WhileInterpreter")
```

Comment:
More explanation for why this is invalid. Same for message below.

Reply:
The while can either be in the body, evaluating the condition, or finished with a terminal env. Exactly one of these should be the case at any given time so one of these options should have a value. If none of them do then there has been a mistake somewhere in the while interpreter's internal state transitions

Reply:
Cool, useful to add this to the error message.

Reply:
Which of the options has a value is, in essence, encoding both the state of the while's FSM and the associated data needed to do further computation